samza-core/src/test/scala/org/apache/samza/system/TestSystemConsumers.scala

```diff
@@ -340,7 +340,7 @@ class TestSystemConsumers {
     assertTrue(consumer.lastPoll.contains(systemStreamPartition1))
   }
 
-  @Test
+  @Test(expected = classOf[Exception])
```

Comment:
Why is this expected to throw an exception?

Reply:
Previously, we checked if the Consumer is a sub-type of StartpointVisitor and now that is removed. This test invokes the default implementation of `SystemConsumer.register(ssp, startpoint)` and hits the `UnsupportedOperationException`. I will improve this test, once we have one consumer which implements both `register(ssp, startpoint)` and `StartpointVisitor`.

Reply:
This class works against the SystemConsumer interface, so this test should be mocking a SystemConsumer, which means you don't need a consumer with a concrete implementation. You should be able to update this test within this PR.

Reply:
1. Just to be clear, the suggestion you're pointing out is not based on the changes done in this patch.
2. Yes. I removed the SystemConsumer implementation present in the test class. Switched to using `InmemorySystemConsumer`.  Using it over mock, since the mock doesn't invoke the default implementation of `SystemConsumer` that throws up the expected exception.

Reply:
This test shouldn't depend on any concrete SystemConsumer, including InmemorySystemConsumer. You don't need a concrete SystemConsumer for this test. You should just have a mock SystemConsumer.

Reply:
Done.